core/src/main/scala/com/microsoft/azure/synapse/ml/causal/LinearDMLEstimator.scala

```diff
@@ -0,0 +1,335 @@
+          case _ => throw new Exception("The defined treatment model does not support weightCol.")
+        }
+        getOutcomeModel match {
+          case w: HasWeightCol => w.set(w.weightCol, getWeightCol)
+          case _ => throw new Exception("The defined outcome model does not support weightCol.")
+        }
+      }
+
+      getTreatmentModel match {
+        case m: HasLabelCol with HasFeaturesCol =>
+          m.set(m.labelCol, getTreatmentCol).set(m.featuresCol, "treatment_features")
+        case _ => throw new Exception("The defined treatment model does not support HasLabelCol and HasFeaturesCol.")
+      }
+
+      getOutcomeModel match {
+        case m: HasLabelCol with HasFeaturesCol =>
+          m.set(m.labelCol, getOutcomeCol).set(m.featuresCol, "outcome_features")
+        case _ => throw new Exception("The defined outcome model does not support HasLabelCol and HasFeaturesCol.")
+      }
+
+      val dmlModel = new LinearDMLModel()
+
+      // sampling with replacement to redraw data and get TE value
+      // Run it for multiple times in parallel, get a number of TE values,
+      // Use average as Ate value, and 2.5% low end, 97.5% high end as Ci value
+      // Create execution context based on $(parallelism)
+      log.info(s"Parallelism: $getParallelism")
+      val executionContext = getExecutionContextProxy
+
+      val ateFutures = Range(1, getMaxIter+1).toArray.map { index =>
+        Future[Double] {
+          log.info(s"Executing ATE calculation on iteration: $index")
+          println(s"Executing ATE calculation on iteration: $index")
+          // sample data with replacement
+          val redrewDF =  if (getMaxIter == 1) dataset else dataset.sample(withReplacement = true, fraction = 1)
+          redrewDF.cache()
+          val ate: Option[Double] =
+            try {
+              val totalTime = new StopWatch
+              val oneAte = totalTime.measure {
+                trainInternal(redrewDF)
+              }
+              println(s"Completed ATE calculation on iteration $index and got ATE value: $oneAte, time elapsed: ${totalTime.elapsed() / 60000000000.0} minutes")
+              Some(oneAte)
+            } catch {
+              case ex: Throwable =>
+                println(s"ATE calculation got exception on iteration $index with the redrew sample data. Exception ignored.")
+                log.info(s"ATE calculation got exception on iteration $index with the redrew sample data. Exception details: $ex")
+                None
+            }
+          redrewDF.unpersist()
+          ate.getOrElse(0.0)
+        }(executionContext)
+      }
+
+      val ates = awaitFutures(ateFutures).filter(_ != 0.0).sorted
+      val finalAte = if (getMaxIter == 1) ates.head else ates.sum / ates.length
+      println(s"Completed $maxIter iteration ATE calculations and got ${ates.length} values, final ATE = $finalAte")
+
+      dmlModel.setAte(finalAte)
```

Comment:
Create the instance here.
```suggestion
      val dmlModel = new LinearDMLModel().setAte(finalAte)
```